File: src/main/java/ru/market/pricing_service/model/entity/Check.java
package ru.market.pricing_service.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import ru.market.hr_service.model.entity.Employee;

import java.time.LocalDate;

@Entity
@Table(name = "products_check")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Check {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer checkId;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    @Column(nullable = false)
    private LocalDate date;
}

File: src/main/java/ru/market/pricing_service/model/entity/Check.kt
package ru.market.pricing_service.model.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import ru.market.hr_service.model.entity.Employee
import java.time.LocalDate

@Entity
data class Check(
    @Id @GeneratedValue val checkId: Int,
    @ManyToOne
    @JoinColumn(name = "employee_id", nullable = false)
    val employee: Employee,
    @Column val date: LocalDate
)